### PR TITLE
GCP/Azure: add `skypilot-user` tags to VMs on these clouds.

### PR DESCRIPTION
**Pull request by concretevitamin**
For better accountability.

**Tested (run the relevant ones):**

GCP head + worker (note: they have "labels" and "tags"; former is used):

Screen Shot 2023-01-14 at 07 48 25

Azure head + worker:

Screen Shot 2023-01-14 at 07 52 49
Screen Shot 2023-01-14 at 07 52 57

Also tested (previously was throwing an error from within optimizer, `No resource satisfying None on Azure`):
```
» sky launch --use-spot -i10 --down --cloud azure
Usage: sky launch [OPTIONS] [ENTRYPOINT]...
Try 'sky launch -h' for help.

Error: SkyPilot currently has not implemented support for spot instances on Azure. Please file an issue if you need this feature.
```

### CODE REVIEW

File: sky/cli.py [hunk 1]
```diff
@@ -1180,6 +1180,15 @@ def launch(
     if backend_name is None:
         backend_name = backends.CloudVmRayBackend.NAME
 
+    # A basic check. Programmatic calls will have a proper (but less
+    # informative) error from optimizer.
+    if (cloud is not None and cloud.lower() == 'azure' and
+            use_spot is not None and use_spot):
```
Comment (WoosukKwon) [inline sky/cli.py:1186]: Can't it be shortened as `use_spot`?

**Review by WoosukKwon — APPROVED** [review]
LGTM. I actually experienced the difficulties in identifying the owner of the VMs in our GCP and Azure accounts. The new update would be really helpful. Thanks for adding this @concretevitamin!

**Comment by concretevitamin** [comment]
Thanks @WoosukKwon. Merging this.
